index.js

```diff
@@ -28,11 +28,13 @@ const pLimit = concurrency => {
 	};
 
 	const enqueue = (fn, resolve, ...args) => {
-		if (activeCount < concurrency) {
-			run(fn, resolve, ...args);
-		} else {
-			queue.push(run.bind(null, fn, resolve, ...args));
-		}
+		queue.push(run.bind(null, fn, resolve, ...args));
+		(async () => {
+			await Promise.resolve();
```

Comment:
Shouldn't this be inside the below `if`-statement? I assume we only want to delay to the next micro-tick if we actually plan to execute the method.

Reply:
I believe this need to happen every time. I tried taking moving it into the `if` and ran into an issue where if 100 `limit` functions are enqueued, the check against `activateCount < concurrency` happens synchronously, but the increment happens asynchronously in the next micro-tick when `run` is called.

The `activeCount` increment could happen here synchronously, but I think the activeCount increment would also need to add incremented just before the `queue.shift()()` in the `next` function and removed from the `run` function.

Reply:
Ok. Can you add a code comment about that? So it’s not accidentally moved into the if-statement in the future.

Reply:
Definitely!